internal/servicebindingcleanup/service_test.go

```diff
@@ -0,0 +1,188 @@
+package servicebindingcleanup_test
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/kyma-project/kyma-environment-broker/internal"
+	"github.com/kyma-project/kyma-environment-broker/internal/broker"
+	"github.com/kyma-project/kyma-environment-broker/internal/servicebindingcleanup"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage/dberr"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage/driver/memory"
+	"github.com/pivotal-cf/brokerapi/v8/domain/apiresponses"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	requestTimeout = 100 * time.Millisecond
+	requestRetries = 2
+)
+
+func TestServiceBindingCleanupJob(t *testing.T) {
+	// setup
+	bindingsStorage := memory.NewBinding()
+	handler := newUnbindHandler(bindingsStorage)
+	srv := newServer(handler)
+	defer srv.Close()
+
+	ctx := context.TODO()
+	brokerClientConfig := broker.ClientConfig{
+		URL: srv.URL,
+	}
+	brokerClient := broker.NewClientWithRequestTimeoutAndRetries(ctx, brokerClientConfig, requestTimeout, requestRetries)
+
+	t.Run("should not unbind service bindings in dry run mode", func(t *testing.T) {
+		// given
+		expectedBinding := internal.Binding{
+			ID:         "binding-id",
+			InstanceID: "instance-id",
+		}
+		require.NoError(t, bindingsStorage.Insert(&expectedBinding))
+
+		svc := servicebindingcleanup.NewService(true, brokerClient, bindingsStorage)
+
+		// when
+		err := svc.PerformCleanup()
+		require.NoError(t, err)
+
+		// then
+		actualBinding, err := bindingsStorage.Get(expectedBinding.InstanceID, expectedBinding.ID)
+		require.NoError(t, err)
+		assert.Equal(t, expectedBinding.ID, actualBinding.ID)
+
+		// cleanup
+		require.NoError(t, bindingsStorage.Delete(expectedBinding.InstanceID, expectedBinding.ID))
+	})
+
+	t.Run("should unbind service bindings", func(t *testing.T) {
```

Comment:
Could we add test case to unbind expired but leave not expired?

Reply:
Yes! Good idea :)